textproc/bfs/Portfile

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+
+name                bfs
+version             1.3.2
+categories          textproc
+platforms           darwin freebsd linux
+license             BSD-0-clause
+maintainers         {@ylluminarious orbitalimpact.com:georgedp} openmaintainer
+description         A breadth-first version of the UNIX find command.
+homepage            https://github.com/tavianator/bfs
+master_sites        ${homepage}/archive/
+distname            ${version}
+
+long_description    bfs is a variant of the UNIX find command that operates breadth-first rather \
+    than depth-first. It is otherwise intended to be compatible with many \
+    versions of find, including: POSIX find, GNU find, \{Free,Open,Net\}BSD find \
+    and macOS find. If you're not familiar with find, the GNU find manual \
+    provides a good introduction.
+
+checksums           rmd160   70cbd792d501f568958c51b65a22cc1cdb0fe17f \
+                    sha256   45939e6c1921c90e5f159407f877178cc9376dc66e808dbef80600d7624cbbc2 \
+                    size     88162
+
+worksrcdir          ${name}-${version}
+use_configure       no
```

Comment:
You are going to need to make sure that you are [using the right compiler](https://trac.macports.org/wiki/UsingTheRightCompiler).

Reply:
@mf2k I don't think this program cares very much about which compiler it uses. It's supposed to be portable, so the default one chosen by Macports should suffice.

Reply:
You need to at least make sure the universal variant works or not. Without the configure phase, manual changes, as mentioned in the wiki page, need to happen to make it work.

Reply:
@mf2k Ok, I added some code for the universal variant so that it can build a universal binary if chosen by the user. [According to the docs:](https://guide.macports.org/chunked/reference.variants.html#reference.variants.user-selected) "The [universal] variant can be overridden if the default code does not work."

Selecting `+universal` does not work on Mojave; it fails with this error:

```
Error: bfs cannot be installed for the configured universal_archs 'x86_64 i386' because it only supports the arch(s) ''.
```

I'm inclined to believe this error occurred because Mojave does not seem to play nice with universal binaries anymore. I don't know whether this warrants a note or not.

However, selecting `+universal` **does** work on High Sierra. It installed successfully and the output of `file /opt/local/bin/bfs` is:

```
/opt/local/bin/bfs: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [i386:Mach-O executable i386]
/opt/local/bin/bfs (for architecture x86_64):	Mach-O 64-bit executable x86_64
/opt/local/bin/bfs (for architecture i386):	Mach-O executable i386
```